integration_tests/src/main/python/misc_expr_test.py

```diff
@@ -31,3 +31,15 @@ def test_part_id():
             lambda spark : unary_op_df(spark, short_gen, num_slices=8).select(
                 f.col('a'),
                 f.spark_partition_id()))
+
+def test_raise_error():
+    data_gen = ShortGen(nullable=False, min_val=0, max_val=20, special_cases=[])
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : unary_op_df(spark, data_gen, num_slices=2).select(
+            f.when(f.col('a') > 30, f.raise_error("unexpected"))))
+
+    assert_gpu_and_cpu_error(
+        lambda spark : unary_op_df(spark, short_gen, num_slices=2).select(
+                f.raise_error(f.col('a'))).collect(),
+        conf={},
+        error_message="java.lang.RuntimeException")
```

Comment:
The test should verify we are properly conveying the specified error message into the exception rather than just checking for the same exception type, checking both the null first element and non-null first element scenarios.

Reply:
Done